bundles/org.corpus_tools.hexatomic.graph/src/main/java/org/corpus_tools/hexatomic/graph/GraphEditor.java

```diff
@@ -0,0 +1,713 @@
+    tblclmnFilterBySegment.setWidth(100);
+    tblclmnFilterBySegment.setText("Filter by segment");
+    graphSash.setWeights(new int[] {300, 100});
+
+    textRangeTable.addSelectionListener(new UpdateViewListener(false));
+    txtSegmentFilter.addModifyListener(new UpdateViewListener(true));
+    btnIncludePointingRelations.addSelectionListener(new UpdateViewListener(true));
+    btnIncludeSpans.addSelectionListener(new UpdateViewListener(true));
+
+    registerGraphControlListeners();
+
+    viewer.getControl().forceFocus();
+
+    updateView(true);
+
+  }
+
+  private void registerGraphControlListeners() {
+
+    // Allow to drag the background area with the mouse
+    GraphDragMoveAdapter.register(viewer.getGraphControl());
+    
+    // Disable the original scroll event when the mouse wheel is activated
+    viewer.getGraphControl().addListener(SWT.MouseVerticalWheel,
+        new org.eclipse.swt.widgets.Listener() {
+
+          @Override
+          public void handleEvent(Event event) {
+            event.doit = false;
+          }
+        });
+
+    // Add a mouse wheel listener that zooms instead of scrolling
+    viewer.getGraphControl().addMouseWheelListener(new MouseWheelListener() {
+
+      @Override
+      public void mouseScrolled(MouseEvent e) {
+
+        ScalableFigure figure = viewer.getGraphControl().getRootLayer();
+        double oldScale = figure.getScale();
+        Optional<Double> newScale = Optional.empty();
+        if (e.count < 0) {
+          newScale = Optional.of(oldScale * 0.75);
+        } else {
+          newScale = Optional.of(oldScale * 1.25);
+        }
+
+        if (newScale.isPresent()) {
+          double clippedScale = Math.max(0.0625, Math.min(2.0, newScale.get()));
+
+          if (clippedScale != oldScale) {
+
+            Point originalViewLocation = viewer.getGraphControl().getViewport().getViewLocation();
+
+            figure.setScale(clippedScale);
+            viewer.getGraphControl().getViewport().validate();
+
+            viewer.getGraphControl().getViewport()
+                .setViewLocation(originalViewLocation.getScaled(clippedScale / oldScale));
+            viewer.getGraphControl().getViewport().validate();
+
+            Point originallyClicked = new Point(e.x, e.y);
+            Point scaledClicked = originallyClicked.getScaled(clippedScale / oldScale);
+            centerViewportToPoint(scaledClicked);
+          }
+        }
+
+      }
+    });
+    // Center the view on the mouse cursor when it was double clicked
+    viewer.getGraphControl().addMouseListener(new MouseListener() {
+
+      @Override
+      public void mouseUp(MouseEvent e) {
+      }
+
+      @Override
+      public void mouseDown(MouseEvent e) {
+      }
+
+      @Override
+      public void mouseDoubleClick(MouseEvent e) {
+        Point clickedInViewport = new Point(e.x, e.y);
+        centerViewportToPoint(clickedInViewport);
+      }
+    });
+
+    // React to arrow keys to move the displayed graph
```

Comment:
```suggestion
    // React to arrow keys to scroll the viewport
```

Think this is better as you don't actually move the graph